modules/transport-netty4/src/main/java/org/opensearch/http/netty4/Netty4HttpServerTransport.java

```diff
@@ -397,7 +397,11 @@ public UpgradeCodec newUpgradeCodec(CharSequence protocol) {
                 handlingSettings.getMaxChunkSize()
             );
 
-            final HttpServerUpgradeHandler upgradeHandler = new HttpServerUpgradeHandler(sourceCodec, upgradeCodecFactory);
+            final HttpServerUpgradeHandler upgradeHandler = new HttpServerUpgradeHandler(
+                sourceCodec,
+                upgradeCodecFactory,
+                handlingSettings.getMaxContentLength()
```

Comment:
Found this one while prototyping https://github.com/opensearch-project/OpenSearch/issues/3000: by default, Netty assumes the upgrade is initiated by GET requests, but in case of OpenSearch, that is not necessarily the case.